### PR TITLE
Layer exclusion naming convention applied to classification nets [DON'T MERGE YET]

### PR DESCRIPTION
**Pull request by gheinrich**
In case someone hits a problem like that mentioned in #601 for a classification network.
Once Caffe implements input layers and phase control from Python we should be able to remove those workarounds.

### CODE REVIEW

File: digits/model/tasks/caffe_train.py [hunk 1]
```diff
@@ -411,7 +411,10 @@ def save_files_classification(self):
                     val_data_layer.hdf5_data_param.batch_size = constants.DEFAULT_BATCH_SIZE
 
         # hidden layers
-        train_val_network.MergeFrom(hidden_layers)
+        for layer in hidden_layers:
+            # don't add layers with names starting with "deploy_" to train/val network
+            if not layer.name.startswith('deploy_'):
+                train_val_network.layer.add().CopyFrom(layer)
```
Comment (lukeyeager) [inline digits/model/tasks/caffe_train.py]: In the `generic` path, I'm stripping the prefix from the final layer name:
https://github.com/NVIDIA/DIGITS/blob/v3.0.0/digits/model/tasks/caffe_train.py#L556

Can we do the same here?

**Comment by lukeyeager** [comment]
I tried using this to integrate my Python accuracy layer ([here](https://gist.github.com/lukeyeager/5ae183044d96ef1540f2)), and got this error:

> ERROR: Layer 'prob' references bottom 'accuracy' at the TEST stage however this blob is not included at that stage. Please consider using an include directive to limit the scope of this layer.

That's because this layer gets added to the deploy network:
```
layer {
  name: "prob"
  type: "Softmax"
  bottom: "accuracy"
  top: "prob"
}
```
The bottom should be `ip2` instead of `accuracy`.

Looking at the code, I don't see a great way around this. I spent some time trying to find a programmatic solution to all this mess a while back ([rough code here](https://github.com/lukeyeager/DIGITS/compare/666e9f2c37ecd297abfccafb7ce2557e9f4cfca9...4dac32c32efdd6410a667ed40f3a72ac89cc8d67)) and I gave up on it because there was simply no way to do it elegantly. I'd love to get https://github.com/BVLC/caffe/issues/1245 integrated and update our default networks so that the networks can be fully, correctly specified without needing to do any magic behind the user's back.

**Comment by gheinrich** [comment]
Updated to strip prefix but as @lukeyeager mentioned there is an issue with the helper code in DIGITS which adds a `softmax` layer to the deploy network.
Needs more work.

**Comment by lukeyeager** [comment]
Closed in favor of https://github.com/NVIDIA/DIGITS/pull/623 (which has been closed in favor of https://github.com/NVIDIA/DIGITS/pull/628 !).

**Pull request closed without merging**